### DISCUSSION TITLE
A permanent staging environment domain.

### DISCUSSION THREAD

**Discussion by yassinebridi** [vercel/vercel#4864] [Ideas] [answered]
Vercel has a pretty powerful deployment and promoting system, between prod and preview.
I'm not sure if a preview is meant to be a staging environment where i can test things before promoting to production.
The problem with that is, the only static domain that is provided when pushing a non-production deployment: ```
```
projectname.accountname.vercel.app
``` 
That domain, is promoted to production, once i push a production deployment: `vercel --prod`.
So i have no way to look back to preview non-prod deployment, especially that i need to configure `CORS`, for each domain(prod, and the static preview domain).

I think it would be pretty useful, if we have another deployment environment `staging`, like `preview` and `prod`, which has a static domain.

**Comment by styfle** [comment, marked as answer]
Hi @yassinebridi 

You can map a git branch to a custom domain in your Project Settings.

https://vercel.com/blog/branch-domains

Does that solve your problem?

**Reply by yassinebridi** [reply]
This definitely solved my issue, thanks @styfle.

**Reply by styfle** [reply]
Just following up here, you can now add Environment Variables to a git branch as well. 

See the announcement: https://vercel.com/changelog/environments-variables-per-git-branch

Also checkout this guide: https://vercel.com/support/articles/set-up-a-staging-environment-on-vercel

**Reply by kasvith** [reply]
@styfle we are not using git branch deployments, we deploy as a part of our CI pipeline
How we can do the same with just CLI?

**Reply by tranquyetdev** [reply]
We are also using Github Action to deploy Nextjs in NX monorepo style, how can we get a static Staging domain? As in NX, we won't create a "staging" branch
